File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        num_set = set(nums)

        max_len = 0
        for num in num_set:
            if num - 1 not in num_set:
                count = 1
                while num + 1 in num_set:
                    num += 1
                    count += 1
                max_len = max(max_len, count)

        return max_len

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        unordered_set<int> st;
        for (int x : nums) {
            st.insert(x);
        }
        int maxLen = 0;
        for (auto x : st) {
            if (st.find(x - 1) == st.end()) {
                int count = 1;
                int num = x;
                while (st.find(num+1) != st.end()) {
                    num++;
                    count++;
                }
                maxLen = max(maxLen, count);
            }
        }
        return maxLen;
    }
};